routing/pathfind.go

```diff
@@ -584,12 +578,15 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 
 	// To start, our target node will the sole item within our distance
 	// heap.
-	heap.Push(&nodeHeap, distance[target])
+	heap.Push(&nodeHeap, &nodeWithDist{
+		node: target,
+		dist: 0,
+	})
```

Comment:
If you don't want the map lookup here, it may be better to store the target struct in a variable. To prevent initializing the same `nodeWithDist` twice.